pkg/algorithm/hived_algorithm_test.go

```diff
@@ -960,6 +986,36 @@ func testBadNodes(t *testing.T, configFilePath string) {
 	}
 }
 
+func testSplitAlloc(t *testing.T, configFilePath string) {
+	sConfig := api.NewConfig(api.InitRawConfig(&configFilePath))
+	(*sConfig.VirtualClusters)["VC1"].VirtualCells[0].CellNumber = 4
+	(*sConfig.VirtualClusters)["VC1"].VirtualCells[3].CellNumber = 0
+	h := NewHivedAlgorithm(sConfig)
+	for _, chains := range h.cellChains {
+		sortChains(chains)
+	}
+	setHealthyNodes(h)
+	allocatedPods = []*core.Pod{}
+
+	pod := allPods["pod44"]
+	pod.Annotations[api.AnnotationKeyPodSchedulingSpec] = common.ToYaml(pss[pod.UID])
+	psr := h.Schedule(pod, []string{"0.0.3.2", "0.0.3.3", "0.0.4.2", "0.0.4.3"}, internal.PreemptingPhase)
+	bindingPod := internal.NewBindingPod(pod, psr.PodBindInfo)
+	h.AddAllocatedPod(bindingPod)
+	allocatedPods = append(allocatedPods, bindingPod)
+	compareSchedulingResult(t, pod, psr)
+
+	h.setBadNode("0.0.3.3")
+	pod = allPods["pod45"]
+	pod.Annotations[api.AnnotationKeyPodSchedulingSpec] = common.ToYaml(pss[pod.UID])
+	psr = h.Schedule(pod, []string{"0.0.3.2", "0.0.3.3", "0.0.4.2", "0.0.4.3"}, internal.PreemptingPhase)
+	if psr.PodBindInfo == nil {
+		t.Errorf("Cannot split higher level cells when requested level cell is bad")
+	} else {
+		compareSchedulingResult(t, pod, psr)
+	}
+}
```

Comment:
how about adding a case where it could not split but it does?

Reply:
added